src/Maestro/SubscriptionActorService/PullRequestDescriptionBuilder.cs

```diff
@@ -0,0 +1,204 @@
+using Maestro.Data.Models;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using static SubscriptionActorService.PullRequestActorImplementation;
+
+namespace SubscriptionActorService
+{
+    public class PullRequestDescriptionBuilder
+    {
+        /// <param name="loggerFactory">Logger factory</param>
+        /// <param name="description">A string containing initialized PR description in case of a new PR, and an existing PR description
```

Comment:
What does initialized PR description mean?

Reply:
In case of a new PR the CreatePullRequestAsync Adds some lines to the description ("initializes it") and then passes it to CommitUpdateAsync which calls the PRBuilder

Reply:
Do you think it makes sense to make the description an optional parameter to the constructor?

* If the description is null or empty, initialize it with the introductory text.
* If it already has something, the class assumes it's an already properly formatted string. 

It seems a little weird that this class' responsibility is to create pull request descriptions, but it needs a string that already represents a pr description.

Reply:
This would allow us to move https://github.com/dotnet/arcade-services/pull/1774/files#diff-c5bf57a757adcea620af121a78f82cd95fec509a1de1ebc7f63a4a7900d3d02aR757-R759 to the constructor for this class.